utils/s2n_safety.h

```diff
@@ -133,25 +133,32 @@
  */
 #define ENSURE_PTR_NONNULL( x )                     ENSURE_PTR((x) != NULL, S2N_ERR_NULL)
 
+#ifdef __TIMING_CONTRACTS__
```

Comment:
Was this supposed to be added back?

Reply:
@camshaft Yes, the `s2nSidetrail` proofs need the `__TIMING_CONTRACTS__` macro to avoid the inclusion of the pre- and post-conditions.

Reply:
All of the `s2n_safety.h` macros are derived from  [`__S2N_ENSURE`](https://github.com/awslabs/s2n/blob/master/utils/s2n_ensure.h#L30). This is replaced with a specific [implementation for sidetrail](https://github.com/awslabs/s2n/blob/master/tests/sidetrail/working/stubs/s2n_ensure.h#L28), so we shouldn't need the `ifdefs`, which I'd like to avoid as much as possible.

If we would prefer the `{POST,PRE}CONDITION` macros work differently than what `__S2N_ENSURE` provides (it calls `__VERIFIER_assume((cond))`), we can expose another hook (something like `__S2N_ENSURE_CONDITION`) that returns `void 0`.

Reply:
@camshaft I implemented as you suggested, could you confirm if it's ok?